arviz/stats/density_utils.py

```diff
@@ -946,20 +947,37 @@ def get_bins(values):
     It is considered a robusts version of the Scott rule as the IQR is less affected by outliers
     than the standard deviation. However, the IQR depends on fewer points than the standard
     deviation, so it is less accurate, especially for long tailed distributions.
+
+    -------
+    Adding functionality for non integer data
+
+    Segment only works if input dtype is not int32. Width of bins is computed with
+    numpy.histogram_bin_edges with bins='auto'. Auto mode uses the maximum of Sturges and
+    Freedman-Diaconis estimators to provide bin margins. Rest of the parameters to
+    numpy.histogram_bin_edges are default.
+
+
     """
-    x_min = values.min().astype(int)
-    x_max = values.max().astype(int)
 
-    # Sturges histogram bin estimator
-    bins_sturges = (x_max - x_min) / (np.log2(values.size) + 1)
+    if values.dtype == "int32":
+        x_min = values.min().astype(int)
+        x_max = values.max().astype(int)
+
+        # Sturges histogram bin estimator
+        bins_sturges = (x_max - x_min) / (np.log2(values.size) + 1)
 
-    # The Freedman-Diaconis histogram bin estimator.
-    iqr = np.subtract(*np.percentile(values, [75, 25]))  # pylint: disable=assignment-from-no-return
-    bins_fd = 2 * iqr * values.size ** (-1 / 3)
+        # The Freedman-Diaconis histogram bin estimator.
+        iqr = np.subtract(
+            *np.percentile(values, [75, 25])
+        )  # pylint: disable=assignment-from-no-return
+        bins_fd = 2 * iqr * values.size ** (-1 / 3)
 
-    width = np.round(np.max([1, bins_sturges, bins_fd])).astype(int)
+        width = np.round(np.max([1, bins_sturges, bins_fd])).astype(int)
 
-    return np.arange(x_min, x_max + width + 1, width)
+        return np.arange(x_min, x_max + width + 1, width)
+
+    else:
+        return histogram_bin_edges(values, bins="auto")
```

Comment:
given that we are using the same approach as `auto` in numpy, I think it would be better to use our `bin_sturges` and `bins_fd`, therefore, if we decided to update the default behaviour, both cases would be updated. Only the `astype` in the min, max and width and the considering 1 in the call to max differ between int and float, the rest are the same.

Reply:
Using the same method as defined in the code where `bin_sturges` and `bin_fd` are calculated with x_min and x_max astype as float. Note that the max possible value for last bin is `x_max + width`. The results are different than `histogram_bin_edges ` function since it places a hard limit on max value of last bin.

https://github.com/rohits5496/arviz/blob/aca1c21d68a8364617fc9c88badf57abe93974ab/arviz/stats/density_utils.py#L977-L979

For example: auto mode on histogram_bin_edges

1. Integer values
Given array : [0,1,1,2,3,5]
get_bins() : [0, 2, 4, 6]
histogram_bin_edges() : [0.  , 1.25, 2.5 , 3.75, 5.  ]

2. Float values
Given array : [0.01, 1,1,2,3,5]
get_bins() : [0.01      , 1.93612423, 3.86224846, 5.78837269]
histogram_bin_edges() : [0.01  , 1.2575, 2.505 , 3.7525, 5.    ]